meta-resin-common/recipes-support/hostapp-update-hooks/files/99-resin-uboot

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+#
+# Script which configures the resinOS_uEnv,txt to use an updated root index
+#
+
+set -o errexit
+
+new_part=$(findmnt --noheadings --canonicalize --output SOURCE "/mnt/sysroot/inactive")
+blockdev=$(basename "$new_part")
+new_part_idx=$(cat "/sys/class/block/$blockdev/partition")
+
+echo "resin_root_part=$new_part_idx" > /mnt/boot/resinOS_uEnv.txt
```

Comment:
This is not atomic - it will first truncate the file before writing the contents.

Reply:
Fixed. Thanks @willnewton